cli/azd/pkg/azapi/stack_deployments.go

```diff
@@ -634,3 +634,124 @@ func convertFromStacksProvisioningState(
 
 	return DeploymentProvisioningState("")
 }
+
+func (d *StackDeployments) ValidatePreflightToResourceGroup(
+	ctx context.Context,
+	subscriptionId string,
+	resourceGroup string,
+	deploymentName string,
+	armTemplate azure.RawArmTemplate,
+	parameters azure.ArmParameters,
+	tags map[string]*string,
+) error {
+	client, err := d.createClient(ctx, subscriptionId)
+	if err != nil {
+		return err
+	}
+
+	templateHash, err := d.CalculateTemplateHash(ctx, subscriptionId, armTemplate)
+	if err != nil {
+		return fmt.Errorf("failed to calculate template hash: %w", err)
+	}
+
+	clonedTags := maps.Clone(tags)
+	clonedTags[azure.TagKeyAzdDeploymentTemplateHashName] = &templateHash
+
+	stackParams := map[string]*armdeploymentstacks.DeploymentParameter{}
+	for k, v := range parameters {
+		stackParams[k] = &armdeploymentstacks.DeploymentParameter{
+			Value: v.Value,
+		}
+	}
+
+	deleteBehavior := armdeploymentstacks.DeploymentStacksDeleteDetachEnumDelete
```

Comment:
FYI @hemarina , @wbreza had merged some changes here. I think we can reuse the `parseDeploymentStackOptions` function for getting the deployment stack options